### DISCUSSION TITLE
Restart check

### DISCUSSION THREAD

**Discussion by icedterminal** [cockpit-project/cockpit#18634] [Ideas]
After every use of the update feature, it prompts for reboot. Instead of always prompting, why not have a check in place to determine if a reboot is necessary?  I know on Debian you can check with `cat /var/run/reboot-required`. If this file exists, reboot. If it doesn't exist, no need to reboot. Could this feature be included?

**Comment by martinpitt** [comment]
Yes, we actually want to include support for Debian's `needrestart` (not just the reboot-required flag). It's also tracked in https://github.com/cockpit-project/cockpit/issues/15267 , but needs someone to find time to implement it. Contributions welcome, of course!
